From patchwork Sat Nov 20 12:23:02 2021
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Kyle Copperfield <kmcopper@danwin1210.me>
X-Patchwork-Id: 12630423
Return-Path: 
 <linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@lists.infradead.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from bombadil.infradead.org (bombadil.infradead.org
 [198.137.202.133])
	(using TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by smtp.lore.kernel.org (Postfix) with ESMTPS id 6EDB8C433EF
	for <linux-rockchip@archiver.kernel.org>;
 Sat, 20 Nov 2021 12:23:53 +0000 (UTC)
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
	d=lists.infradead.org; s=bombadil.20210309; h=Sender:
	Content-Transfer-Encoding:Content-Type:List-Subscribe:List-Help:List-Post:
	List-Archive:List-Unsubscribe:List-Id:MIME-Version:Message-Id:Date:Subject:Cc
	:To:From:Reply-To:Content-ID:Content-Description:Resent-Date:Resent-From:
	Resent-Sender:Resent-To:Resent-Cc:Resent-Message-ID:In-Reply-To:References:
	List-Owner; bh=3/OcZ3UOvg3VkKRmnM7m3wBFj9ypz343GNQvErlRNSs=; b=1RSLEPRAI5U1PW
	HQeXUB+gC/N9LkkbZdZ/uX2aP4C/fhsQXC4wc/KLGQFvElI4G6wbPu7FD19baKLi3Z8WIX37bL1Yj
	oekE9/LXTj3W9fTMLJybDKZCM3ASBTKmqeaPwfSJVghHbL2q0Uo3cHhu5AWY/10y3k+G8CV/ugJRC
	Lgr9RtHG22LVnUS4ep0gpsavphbx2tzEV6VBZR+5TW3/54abi679sxBICH+zyBw8l1Rzaoef2H+nx
	A8GXeDqb0CycqIbKj6BlizRXUeD7hslrvCIHnjobXnmL80qQ6X+3j0J4yqhINa0yRN5SN1c9wGgGr
	PcQT1lnEgff4BG5YuQNg==;
Received: from localhost ([::1] helo=bombadil.infradead.org)
	by bombadil.infradead.org with esmtp (Exim 4.94.2 #2 (Red Hat Linux))
	id 1moPPJ-00CVSZ-FB; Sat, 20 Nov 2021 12:23:49 +0000
Received: from danwin1210.me ([2a01:4f8:c010:d56::1])
 by bombadil.infradead.org with esmtps (Exim 4.94.2 #2 (Red Hat Linux))
 id 1moPP5-00CVQh-Lr; Sat, 20 Nov 2021 12:23:37 +0000
Received: from danwin1210.me (unknown [10.9.0.4])
 (using TLSv1.3 with cipher TLS_AES_256_GCM_SHA384 (256/256 bits)
 key-exchange X448 server-signature ECDSA (P-384)
 client-signature ED448)
 (Client CN "danwin1210.me", Issuer "danwin1210.me" (verified OK))
 by mail.danwin1210.me (Postfix) with ESMTPS id C04D21F4B7;
 Sat, 20 Nov 2021 12:23:33 +0000 (UTC)
Received: from prine.. (unknown [10.9.0.1])
 (using TLSv1.3 with cipher TLS_AES_256_GCM_SHA384 (256/256 bits)
 key-exchange X25519 server-signature ECDSA (P-384) server-digest SHA384)
 (No client certificate requested)
 by danwin1210.me (Postfix) with ESMTPSA id CF0493C73F;
 Sat, 20 Nov 2021 12:23:27 +0000 (UTC)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=danwin1210.me;
 s=mail; t=1637411012;
 h=from:from:reply-to:subject:subject:date:date:message-id:message-id:
 to:to:cc:cc:mime-version:mime-version:
 content-transfer-encoding:content-transfer-encoding;
 bh=1D6TK4pBt+NhUD9V+WvMMJAWVBcNVUjT9h+cnTkVwa4=;
 b=lGqKcqM1w4bzz8fvqwwRkdQlQWLcNE7VjVfY5lBAiE3H0Xng3QMa5IpOz8Gc51LojS1L/B
 gN+b78QHgc4gFNhiFA8yHPIKzGCXjJvS2zAtB0qJdiHLa9saQ38qLkDa1F0TtDgHjItf7t
 G5jiHsnSVLPB7M8TjMAjJdO5KAgUtK7yCqV1Uu4oENhSkd0iNlzR/shwYQqvmFHTAi6Z6/
 ooM2ljJpP3i8CaGaeBVI6hiNNdc9IZpVR17tzjIeE5kKCIT5o4XIv0bkb31EDzzuWu+hTS
 piukyCFnyIxY9dCrWpLspKQ8vtdPvjT4GOu5PFm2CSP6OxxT9JGpuGbs2Rs50rUhUk79C9
 m/5nKxVvnRHWVTv9c3Fyiugzams10SukBr2VmYNG4qMPan4Fy9nnmOyXUFUcJf6t06Kez0
 1dzHJotCm6v8GUDd6SO1vUEGJGoVdm7zDJBMyECyyN+W0UDaLDsU2HwE+Yk0uN8HvAp3R6
 kU3QALwN02tgnaOMbSJLuVHfJWI7UIdOuESaBOOA+EGlAo7zFLlLCTzjXEB/vNeI/ewisR
 Fp3LRNE3Do2IjVcjp4ozCm7V6ThmanxlM86uX5eJ56Y7fZhyaEny7p7RTNNrg44UXWJJKa
 BWkgufRYOOZsgzTPUF+O+V8qmc4WXpM400PO7wNzdnlcEm8iuvuZg=
DKIM-Signature: v=1; a=ed25519-sha256; c=relaxed/relaxed; d=danwin1210.me;
 s=ed25519; t=1637411013;
 h=from:from:reply-to:subject:subject:date:date:message-id:message-id:
 to:to:cc:cc:mime-version:mime-version:
 content-transfer-encoding:content-transfer-encoding;
 bh=1D6TK4pBt+NhUD9V+WvMMJAWVBcNVUjT9h+cnTkVwa4=;
 b=KfAvgOaW3xIq59Ehg6Er/7s4YExuLtnJW/GBF57Bcrdn1nO0kJNqeoTUXk81yuTrWxj4PJ
 siZqwcX5VgRbPHAw==
From: Kyle Copperfield <kmcopper@danwin1210.me>
To: Ezequiel Garcia <ezequiel@collabora.com>,
 Mauro Carvalho Chehab <mchehab@kernel.org>,
 Heiko Stuebner <heiko@sntech.de>, Dan Carpenter <dan.carpenter@oracle.com>
Cc: Kyle Copperfield <kmcopper@danwin1210.me>,
 Dragan Simic <dragan.simic@gmail.com>, linux-media@vger.kernel.org,
 linux-rockchip@lists.infradead.org, linux-arm-kernel@lists.infradead.org,
 linux-kernel@vger.kernel.org
Subject: [PATCH] media: rockchip/rga: do proper error checking in probe
Date: Sat, 20 Nov 2021 12:23:02 +0000
Message-Id: <20211120122321.20253-1-kmcopper@danwin1210.me>
MIME-Version: 1.0
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20211120_042335_899153_0C69E4B3 
X-CRM114-Status: GOOD (  13.64  )
X-BeenThere: linux-rockchip@lists.infradead.org
X-Mailman-Version: 2.1.34
Precedence: list
List-Id: Upstream kernel work for Rockchip platforms
 <linux-rockchip.lists.infradead.org>
List-Unsubscribe: <http://lists.infradead.org/mailman/options/linux-rockchip>,
 <mailto:linux-rockchip-request@lists.infradead.org?subject=unsubscribe>
List-Archive: <http://lists.infradead.org/pipermail/linux-rockchip/>
List-Post: <mailto:linux-rockchip@lists.infradead.org>
List-Help: <mailto:linux-rockchip-request@lists.infradead.org?subject=help>
List-Subscribe: <http://lists.infradead.org/mailman/listinfo/linux-rockchip>,
 <mailto:linux-rockchip-request@lists.infradead.org?subject=subscribe>
Sender: "Linux-rockchip" <linux-rockchip-bounces@lists.infradead.org>
Errors-To: 
 linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@lists.infradead.org

The latest fix for probe error handling contained a typo that causes
probing to fail with the following message:

  rockchip-rga: probe of ff680000.rga failed with error -12

This patch fixes the typo.

Fixes: e58430e1d4fd (media: rockchip/rga: fix error handling in probe)
Reviewed-by: Dragan Simic <dragan.simic@gmail.com>
Signed-off-by: Kyle Copperfield <kmcopper@danwin1210.me>
Reviewed-by: Kieran Bingham <kieran.bingham+renesas@ideasonboard.com>
Reviewed-by: Dan Carpenter <dan.carpenter@oracle.com>
---
 drivers/media/platform/rockchip/rga/rga.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/media/platform/rockchip/rga/rga.c b/drivers/media/platform/rockchip/rga/rga.c
index 6759091b15e0..d99ea8973b67 100644
--- a/drivers/media/platform/rockchip/rga/rga.c
+++ b/drivers/media/platform/rockchip/rga/rga.c
@@ -895,7 +895,7 @@ static int rga_probe(struct platform_device *pdev)
 	}
 	rga->dst_mmu_pages =
 		(unsigned int *)__get_free_pages(GFP_KERNEL | __GFP_ZERO, 3);
-	if (rga->dst_mmu_pages) {
+	if (!rga->dst_mmu_pages) {
 		ret = -ENOMEM;
 		goto free_src_pages;
 	}
